### PR TITLE
Make --print default to Reason

### PR DESCRIPTION
**Pull request by tekknolagi**
Per @chenglou's request, `refmt` should default to printing Reason code, not binary. This is a saner default for people trying out the `refmt`.

/cc @yunxing

### CODE REVIEW

**Comment by yunxing** [comment]
Looks nice. Thanks! 

I like how cmdliner makes this much easier.

**Comment by tekknolagi** [comment]
Yep! Looking into the failing tests now.

File: miscTests/jsxPpxTest.sh [hunk 1]
```diff
@@ -9,7 +9,7 @@ do
 
   expected=`cat $testPath/expected$i.re`
 
-  ocamlc -dsource -ppx ./reactjs_jsx_ppx.native -pp ./refmt_impl.native -impl $test \
+  ocamlc -dsource -ppx ./reactjs_jsx_ppx.native -pp "./refmt_impl.native --print binary" -impl $test \
```
Comment (yunxing) [inline miscTests/jsxPpxTest.sh]: I think you need to do the same for some rules in rebuild as well.

**Comment by yunxing** [comment]
This diff looks good to me. 

@jordwalke Do you mind help to also review this?

**Pull request merged**